fe1-web/src/features/notification/screens/NotificationScreen.tsx

```diff
@@ -0,0 +1,149 @@
+import { useNavigation } from '@react-navigation/core';
+import { StackScreenProps } from '@react-navigation/stack';
+import React, { useMemo } from 'react';
+import {
+  FlatList,
+  StyleSheet,
+  Text,
+  TextStyle,
+  TouchableOpacity,
+  View,
+  ViewStyle,
+} from 'react-native';
+import { Button } from 'react-native-elements';
+import { useSelector } from 'react-redux';
+
+import ScreenWrapper from 'core/components/ScreenWrapper';
+import { dispatch } from 'core/redux';
+import { Typography } from 'core/styles';
+import STRINGS from 'resources/strings';
+
+import { NotificationHooks } from '../hooks';
+import { NotificationStackParamList } from '../navigation/NotificationStackParamList';
+import { discardNotifications, NotificationState, makeAllNotificationsSelector } from '../reducer';
+
+interface ListSeparatorItem {
+  title: string;
+}
+
+interface NotificationItem extends NotificationState {
+  isLastItem: boolean;
+}
+
+const NotificationScreenStyles = StyleSheet.create({
+  notificationItem: {
+    paddingTop: 16,
+    paddingBottom: 16,
+    borderBottomColor: '#000',
+    borderBottomWidth: 1,
+  } as ViewStyle,
+
+  lastNotificationItem: {
+    borderBottomWidth: 0,
+  } as ViewStyle,
+});
+
+type NavigationProps = StackScreenProps<
+  NotificationStackParamList,
+  typeof STRINGS.notification_navigation_tab_notifications
+>;
+
+const NotificationScreen = () => {
+  const navigation = useNavigation<NavigationProps['navigation']>();
+
+  const laoId = NotificationHooks.useCurrentLaoId();
+  const selectAllNotifications = useMemo(
+    () => makeAllNotificationsSelector(laoId.valueOf()),
+    [laoId],
+  );
+  const notifications = useSelector(selectAllNotifications);
+  const notificationData = useMemo(() => {
```

Comment:
if possible, I'd try to extract a function out of this (and out of the component), as it seems very much independent of everything except for the `notifications` input.
It'd also clarify the goal of the code, as having `notificationData = useMemo(() => {....` doesn't tell me anything about what you're trying to achieve here.

Reply:
I hope it is clearer now? 😅